zerver/views/users.py

```diff
@@ -169,10 +163,7 @@ def patch_bot_backend(
         default_events_register_stream: Optional[str]=REQ(default=None),
         default_all_public_streams: Optional[bool]=REQ(default=None, validator=check_bool)
 ) -> HttpResponse:
-    try:
-        bot = get_user_profile_by_id_in_realm(bot_id, user_profile.realm)
-    except UserProfile.DoesNotExist:
-        return json_error(_('No such user'))
+    bot = access_user_by_id(user_profile, bot_id)
```

Comment:
I think most of these functions should be calling an `access_bot_by_id`, that does the "not a bot" and `can_admin_user` checks as well.

And this should go in `zerver/lib/users.py`.